### ISSUE TITLE
syntax-highlight of hotkey

### ISSUE THREAD

**Issue by CoffeeChaton** [helsmy/vscode-autohotkey#18]
```jsonc
"hotkey_raw": {
    "name": "meta.hotkey.line.ahk",
    "begin": "^([#!^+&<>*~$]+|(?:<\\^>!))?([!-/]|[:-@]|[\\[-`]|[\\{-~]|[a-zA-Z0-9]+)(\\s&\\s([!-/]|[:-@]|[\\[-`]|[\\{-~]|[a-zA-Z0-9]+))?(?=::)",
    "beginCaptures": {
        "1": { "name": "storage.modifier.hotkey.ahk" },
        "2": { "patterns": [ { "include": "#keylist" } ] },
        "4": { "patterns": [ { "include": "#keylist" } ] }
    },
    "end": "::",
    "endCaptures": { "0": { "name": "punctuation.definition.end.hotkey.ahk" } },
    "patterns": [ { "include": "#comment" } ]
},
"hotkey": {
    "name": "meta.hotkey.line.ahk",
    "begin": "^([#!^+&<>*~$]+|(?:<\\^[<>]!))?([!-/]|[:-@]|[\\[-`]|[\\{-~]|[a-zA-Z0-9]+)([ \\t]&[ \\t]([!-/]|[:-@]|[\\[-`]|[\\{-~]|[a-zA-Z0-9]+))?([ \\t]+(?i:UP))?(?=::)",
    //                                                                                                                                          ^1 support of "UP" suffix
    "beginCaptures": {
        "1": { "name": "storage.modifier.hotkey.ahk" },
        "2": {
            "name": "meta.2.hotkey.line.ahk",
            "patterns": [
                { "include": "#keylist" },
                {
                    "match": "[!-/]|[:-@]|[\\[-`]|[\\{-~]",
                    "name": "string.2.storage.modifier.hotkey.ahk"
                    //       ^2 string avoid user open "editor.matchBrackets", then {} () [] <> may be out of control
                },
                {
                    "match": "(?<![a-zA-Z0-9])[a-zA-Z0-9](?![a-zA-Z0-9])",
                    //       ^3 Highlight single-character keywords, which can hint the user to accidentally enter an invalid key
                    "name": "markup.bold.hotkey.ahk"
                }
            ]
        },
        "4": {
            "name": "meta.4.hotkey.line.ahk",
            "patterns": [
                { "include": "#keylist" },
                {
                    "match": "[!-/]|[:-@]|[\\[-`]|[\\{-~]",
                    "name": "string.2.storage.modifier.hotkey.ahk"
                },
                {
                    "match": "(?<![a-zA-Z0-9])[a-zA-Z0-9](?![a-zA-Z0-9])",
                    "name": "markup.bold.hotkey.ahk"
                }
            ]
        },
        "5": { "name": "storage.modifier.up.hotkey.ahk" }
    },
    "end": "::",
    "patterns": [ { "include": "#comment" } ]
},
```

## ^1 support of "UP" suffix

At the end of the hotkey, using up as the suffix is valid and can be run.

```ahk
F10 & F11 Up::
    MsgBox, % A_ThisHotkey
return
```

## ^2 avoid "editor.matchBrackets" out of control

string avoid user open ["editor.matchBrackets"](https://code.visualstudio.com/blogs/2021/09/29/bracket-pair-colorization), then {} () [] <> may be out of control

![2](https://user-images.githubusercontent.com/63182963/231497410-9fdf94ca-ea9f-4e3e-8482-2706e8d3c4bf.png)

## ^3 Highlight single-character keywords

If all valid keys are highlighted, the invalid keys will be very obvious,
So I suggest highlighting `[a-zA-Z0-9]{1}`

![3](https://user-images.githubusercontent.com/63182963/231497265-9d505dc2-3473-4599-8357-5dd467d37d54.png)

---

I am not good at yaml, so I can only provide jsonc for your reference

**Comment by CzarOfScripts** [comment]
Have you tried the `vk**` code?

**Comment by CoffeeChaton** [comment]
> Have you tried the code?`vk**`

keylist is cover.

```json
{ "include": "#keylist" },
```

 <https://github.com/helsmy/vscode-autohotkey/blob/2315db78904ab325f47bcdeaa897600680c8fb5d/syntaxes/ahk.tmLanguage.json#L306>

**Comment by helsmy** [comment]
Thank for your suggestion. It is excellent. 
### `UP` match.
Great.let's merge it 
![图片](https://user-images.githubusercontent.com/47475699/232221207-a552f631-38f3-47da-8151-7be34192a944.png)
### Additional patterns for single chararcter keys. 
Great, let's merge it, too.
![图片](https://user-images.githubusercontent.com/47475699/232221437-8c93f25f-9e3c-40e5-b76b-67dc38e3725c.png)
### Name of brackets. 
The color of `string` is not always the same with `markup`. It depends on themes.
![图片](https://user-images.githubusercontent.com/47475699/232220330-cec50b52-539e-4515-a22c-31d0cfd8635a.png)
Some themes also provide `bold` effect for `bold` name section and I think it is better for hotkey. But, In the theme I used (Dark+), `string.bold` is not effective. We may need try some other options.  
Here is some available name: [naming-conventions](https://macromates.com/manual/en/language_grammars#naming-conventions)

**Comment by helsmy** [comment]
`string` and `markup` section can take effect at the same time. So, it can be fixed #19. New highlighting will be added in the next version.
